packages/jbrowse-plugin-apollo/src/ApolloTextSearchAdapter/ApolloTextSearchAdapter.ts

```diff
@@ -0,0 +1,72 @@
+import { readConfObject } from '@jbrowse/core/configuration'
+import {
+  BaseAdapter,
+  BaseTextSearchAdapter,
+  BaseTextSearchArgs,
+} from '@jbrowse/core/data_adapters/BaseAdapter'
+import BaseResult from '@jbrowse/core/TextSearch/BaseResults'
+import { UriLocation } from '@jbrowse/core/util'
+import { getFetcher } from '@jbrowse/core/util/io'
+
+export class ApolloTextSearchAdapter
+  extends BaseAdapter
+  implements BaseTextSearchAdapter
+{
+  get baseURL() {
+    return readConfObject(this.config, 'baseURL').uri
+  }
+
+  get trackId() {
+    return readConfObject(this.config, 'trackId').uri
+  }
+
+  get internetAccountPreAuthorization():
+    | { authInfo: { token: string }; internetAccountType: string }
+    | undefined {
+    return readConfObject(this.config, 'baseURL')
+      .internetAccountPreAuthorization
+  }
+
+  async searchFeatures(term: string) {
+    const url = new URL(`features/searchFeatures?term=${term}`, this.baseURL)
+    const uri = url.toString()
+    const location: UriLocation = { locationType: 'UriLocation', uri }
+    if (this.internetAccountPreAuthorization) {
+      location.internetAccountPreAuthorization =
+        this.internetAccountPreAuthorization
+    }
+    const fetch = getFetcher(location, this.pluginManager)
+    return fetch(uri)
+      .then((res) => res.json())
+      .catch((err) => err)
+  }
+
+  mapBaseResult(
+    features: {
+      refSeq: any // eslint-disable-line @typescript-eslint/no-explicit-any
+      start: number
+      end: number
+    }[],
+    query: string,
+  ) {
+    return features.map(
+      (feature) =>
+        new BaseResult({
+          label: query,
+          displayString: query,
+          trackId: this.trackId,
+          locString: `${feature.refSeq?.name}:${feature.start}..${feature.end}`,
+        }),
+    )
+  }
+
+  searchIndex(args: BaseTextSearchArgs): Promise<BaseResult[]> {
+    const query = args.queryString
+    return this.searchFeatures(query)
+      .then((features) => this.mapBaseResult(features, query))
+      .catch((err) => err)
```

Comment:
This should be in async-await style for consistency with existing code.
```suggestion
  async searchIndex(args: BaseTextSearchArgs): Promise<BaseResult[]> {
    console.log({ args })
    const query = args.queryString
    const features = await this.searchFeatures(query)
    return this.mapBaseResult(features, query)
```